scylla/src/transport/session.rs

```diff
@@ -234,6 +234,8 @@ pub struct SessionConfig {
 pub enum KnownNode {
     Hostname(String),
     Address(SocketAddr),
+    #[cfg(feature = "cloud")]
+    CloudHostnameAndDC(String, String),
```

Comment:
If we want to have a variant in a public enum which is conditionally enabled depending on a feature, then I think the enum has to be made `#[non_exhaustive]`. As far as I know, if B depends on A then B can't really check whether A has some feature enabled (unless B explicitly enables it) because some other crate C which also depends on A may enable it. In order not to break B when C is added to the project, let's make the enum non-exhaustive.